readthedocs/core/static-src/core/js/doc-embed/sponsorship.js

```diff
@@ -0,0 +1,186 @@
+/* Read the Docs - Documentation promotions */
+
+var constants = require('./constants'),
+    rtddata = require('./rtd-data');
+
+var rtd;
+
+function init() {
+    var post_data = {},
+        force,
+        params;
+
+    rtd = rtddata.get();
+
+    if (!rtd.show_promo()) {
+        return;
+    }
+
+    post_data.placements = get_placements(rtd);
+    post_data.project = rtd.project;
+
+    if (typeof URL !== 'undefined' && typeof URLSearchParams !== 'undefined') {
```

Comment:
Where do these variables come from? They look like new JS things that will not work on older browsers? https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams

Reply:
That is correct. They will not. Hence the `typeof` checking. Old browsers will not be able to force a specific promo or campaign. However, with the `typeof` checking they will not trigger errors.

Reply:
Gotcha. I do worry about using newer browser features for something whose primary purpose is linking to others, because it might break on their browser. We can change it if it becomes an issue though.

Reply:
Unfortunately reading GET params without these builtins or adding a purpose built library becomes somewhat non-trivial.

Reply:
`URLSearchParams` is supported everywhere but older IE/Edge versions. This is ~3% of our traffic overall. I did verify this code path on IE11 to make sure there isn't an issue.

https://caniuse.com/#feat=urlsearchparams